### PR TITLE
DAOS-8887 dtx: trigger DTX aggregation properly

### PR DESCRIPTION
**Pull request by Nasf-Fan**
If the whole pool committed DTX entries count exceeds the threshold,
then even if every container committed DTX entries does not exceeds
the threshold, we still need to choose at least one container to do
DTX aggregation to reduce the DRAM/space pressure.

Allow-unstable-test: true

Signed-off-by: Fan Yong <fan.yong@intel.com>

### CODE REVIEW

**Review by jolivier23 — DISMISSED** [review]

File: src/dtx/dtx_common.c [hunk 1]
```diff
@@ -300,15 +297,16 @@ dtx_aggregate(void *arg)
...
+		     dtx_hlc_age2sec(stat.dtx_first_cmt_blob_time_lo) <= dtx_agg_thd_age_lo))
 			break;
 	}
 
 	sched_req_put(dbca->dbca_agg_req);
 	dbca->dbca_agg_req = NULL;
 
 out:
+	D_ASSERT(dbca->dbca_pool->dbpa_aggregating != 0);
+	dbca->dbca_pool->dbpa_aggregating--;
```
Comment (NiuYawei) [inline src/dtx/dtx_common.c:309]: I don't quite follow this, this ULT is tracked by 'dbca_agg_req', how can multiple ULTs co-exists? (I see you changed dbpa_aggregating from bool to an integer).

Comment (Nasf-Fan) [reply src/dtx/dtx_common.c:309]: Each container can have one ULT to do DTX aggregation on the container. The pool may contain multiple containers.

Comment (NiuYawei) [reply src/dtx/dtx_common.c:309]: oh, I see. dbca_agg_req is per vos cont but dpca_aggregating is per vos pool. Thanks.

**Review by NiuYawei — APPROVED** [review]

**Comment by Nasf-Fan** [comment]
test_create_pool_quantity failed for DAOS-9038.

**Review by NiuYawei — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Comment by Nasf-Fan** [comment]
It is replaced by the patch https://github.com/daos-stack/daos/pull/7447

**Pull request closed without merging**